src/bootstrap/bin/rustc.rs

```diff
@@ -170,6 +170,19 @@ fn main() {
         let crate_name = maybe_crate_name.unwrap();
         maybe_crate = Some(crate_name);
 
+        // librand.rlib would be linked into RTP executable in which only IE mode is supported
+        if env::var_os("USE_TLS_IE_ON_RAND").is_some() {
```

Comment:
Is this env var needed in conjunction with the check for the vxworks target? Does this codegen flag always need to be passed in for `rand` on vxworks?